packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts

```diff
@@ -0,0 +1,446 @@
+  }
+
+  /**
+   * Render the actions in this chain
+   */
+  public renderActions(): CfnListener.ActionProperty[] {
+    return this.renumber([this.actionJson, ...this.next?.renderActions() ?? []]);
+  }
+
+  /**
+   * Called when the action is being used in a listener
+   */
+  public bindToListener(scope: Construct, listener: IApplicationListener, associatingConstruct?: IConstruct) {
+    // Empty on purpose
+    Array.isArray(scope);
+    Array.isArray(listener);
+    Array.isArray(associatingConstruct);
+  }
+
+  /**
+   * Renumber the "order" fields in the actions array.
+   *
+   * We don't number for 0 or 1 elements, but otherwise number them 1...#actions
+   * so ELB knows about the right order.
+   *
+   * Do this in `ListenerAction` instead of in `Listener` so that we give
+   * users the opportunity to override by subclassing and overriding `renderActions`.
+   */
+  protected renumber(actions: CfnListener.ActionProperty[]): CfnListener.ActionProperty[] {
+    if (actions.length < 2) { return actions; }
+
+    return actions.map((action, i) => ({ ...action, order: i + 1 }));
+  }
+}
+
+/**
+ * Options for `ListenerAction.forward()`
+ *
+ * @experimental
+ */
+export interface ForwardOptions {
+  /**
+   * For how long clients should be directed to the same target group
+   *
+   * Range between 1 second and 7 days.
+   *
+   * @default - No stickiness
+   */
+  readonly stickinessDuration?: Duration;
+}
+
+/**
+ * A Target Group and weight combination
+ *
+ * @experimental
+ */
+export interface WeightedTargetGroup {
+  /**
+   * The target group
+   */
+  readonly targetGroup: IApplicationTargetGroup;
+
+  /**
+   * The target group's weight
+   *
+   * Range is [0..1000).
+   *
+   * @default 1
+   */
+  readonly weight?: number;
+}
+
+/**
+ * Options for `ListenerAction.fixedResponse()`
+ *
+ * @experimental
+ */
+export interface FixedResponseOptions {
+  /**
+   * Content Type of the response
+   *
+   * Valid Values: text/plain | text/css | text/html | application/javascript | application/json
+   *
+   * @default - Automatically determined
+   */
+  readonly contentType?: string;
+
+  /**
+   * The response body
+   *
+   * @default - No body
+   */
+  readonly messageBody?: string;
+}
+
+/**
+ * Options for `ListenerAction.redirect()`
+ *
+ * A URI consists of the following components:
+ * protocol://hostname:port/path?query. You must modify at least one of the
+ * following components to avoid a redirect loop: protocol, hostname, port, or
+ * path. Any components that you do not modify retain their original values.
+ *
+ * You can reuse URI components using the following reserved keywords:
+ *
+ * - `#{protocol}`
+ * - `#{host}`
+ * - `#{port}`
+ * - `#{path}` (the leading "/" is removed)
+ * - `#{query}`
+ *
+ * For example, you can change the path to "/new/#{path}", the hostname to
+ * "example.#{host}", or the query to "#{query}&value=xyz".
+ *
+ * @experimental
+ */
+export interface RedirectOptions {
```

Comment:
Would be useful to provide a property called `url` and parses out these, besides having to define each of these explicitly.